dbt/include/spark/macros/adapters.sql

```diff
@@ -27,30 +50,64 @@
   {%- endif %}
 {%- endmacro -%}
 
+
+{% macro clustered_cols(label, required=false) %}
+  {%- set cols = config.get('clustered_by', validator=validation.any[list, basestring]) -%}
+  {%- set buckets = config.get('buckets', validator=validation.any[int]) -%}
+  {%- if (cols is not none) and (buckets is not none) %}
+    {%- if cols is string -%}
+      {%- set cols = [cols] -%}
+    {%- endif -%}
+    {{ label }} (
+    {%- for item in cols -%}
+      {{ item }}
+      {%- if not loop.last -%},{%- endif -%}
+    {%- endfor -%}
+    ) into {{ buckets }} buckets
+  {%- endif %}
+{%- endmacro -%}
+
+
 {% macro spark__create_table_as(temporary, relation, sql) -%}
   {% if temporary -%}
     {{ spark_create_temporary_view(relation, sql) }}
   {%- else -%}
     create table {{ relation }}
     {{ file_format_clause() }}
     {{ partition_cols(label="partitioned by") }}
+    {{ clustered_cols(label="clustered by") }}
+    {{ location_clause() }}
+    {{ comment_clause() }}
     as
       {{ sql }}
   {%- endif %}
 {%- endmacro -%}
 
+
 {% macro spark__create_view_as(relation, sql) -%}
-  create view {{ relation }} as
+  create view {{ relation }}
+  {{ comment_clause() }}
+  as
     {{ sql }}
 {% endmacro %}
 
+
 {% macro spark__get_columns_in_relation(relation) -%}
   {% call statement('get_columns_in_relation', fetch_result=True) %}
       describe {{ relation }}
   {% endcall %}
 
-  {% set table = load_result('get_columns_in_relation').table %}
-  {{ return(sql_convert_columns_in_relation(table)) }}
+  {% set columns = [] %}
```

Comment:
Other folks (#39, #46) are doing similar processing in python rather than Jinja to handle the extraneous metadata rows. I think that approach makes more sense.

Reply:
I'm not sure how to actually do that, should I implement the method in impl.py and then call `{{ do adapter....}}` from here?

Reply:
The approach I'm seeing is to leave the Jinja/SQL of `spark__get_columns_in_relation` to just calling the statement with `describe {{ relation }}` (or `describe {{ relation }} extended`). Then the logic for filtering out pseudo-rows is implemented in a Python method called within `get_catalog`, e.g. [here](https://github.com/fishtown-analytics/dbt-spark/pull/39/files#diff-8a1a92f4d3b6bd5f67adf036a783869fR135).

The logic you've included is definitely an improvement over the current behavior. I think—correct me if I'm wrong—it's most closely related to docs generation, and outside the scope of the changes for this PR. We're definitely going to ship something like it in the next release.

Reply:
@NielsZeilemaker Could you remove this code? I think we'll want to do something similar, just not in the current PR.

Reply:
Done